### ISSUE TITLE
false positive errors when calling operator override that returns `Never` from another function that returns `Never`

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#800]
Code sample in [basedpyright playground](https://basedpyright.com/?typeCheckingMode=all&reportImplicitStringConcatenation=false&reportUnusedExpression=false&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoByApgG5EgCwAUNdQMYA2AhgM4tQBiYYAXNVAKgATIsCgB9cRACuDSQAoWRBsAA0UEkwbSiPKMAZgmMdQHoAlFAC0APkKlyfKoJdQQTJEqgBRAB50iBBgkMBR5ACIZBmCEBiQ6YxCUTHYAIyYUFCIhcPNaZ0ERMUkQKIUlFXVNbV19Q2MzS1t7MhA9AGJOaRQ6YNCoAHckGAALYSJGJhBstyIYaRBk%2BAQiKHCCMAAlOYWUcKgEzLB8afnFtY2s-flpnDxts5QAFUQiPILXKAqxACoNLR0-EEQIERQk4lKsnExhgRGwMHEACZyso1P8anoDEYTFALNY7MRWk5PgJTrsvqioH9qjooJ0ON1eklBsMxiJJtMhLNHrBXhctjtFvtDihjpgetoRFAmNzySwYMY4UQUPgboFcDAAOLK8haF4rACSbB0LHMQA)

```python
from typing import Never


class Foo:
    def __mul__(self, value: float, /) -> Never:
        raise Exception("multiplication is banned")

    def __rmul__(self, value: float, /) -> Never: # Function with declared return type "NoReturn" cannot return "None" (reportReturnType)
        self * value
    
    def __rmul_attempt_2__(self, value: float, /) -> Never:
        return self * value # Function with declared return type "NoReturn" cannot include a return statement (reportGeneralTypeIssues)
```

both of these errors are incorrect, since the `__mul__` method returns `Never`
